File: app/src/main/java/io/smileyjoe/applist/comparator/AppDetailComparator.java
package io.smileyjoe.applist.comparator;

import java.util.Comparator;

import io.smileyjoe.applist.object.AppDetail;

public class AppDetailComparator implements Comparator<AppDetail> {

    @Override
    public int compare(AppDetail appDetailLeft, AppDetail appDetailRight) {
        return appDetailLeft.getName().toLowerCase().compareTo(appDetailRight.getName().toLowerCase());
    }
}

File: app/src/main/java/io/smileyjoe/applist/comparator/AppDetailComparator.kt
package io.smileyjoe.applist.comparator

import io.smileyjoe.applist.objects.AppDetail

/**
 * Sort a list of [AppDetail] alphabetically
 */
class AppDetailComparator : Comparator<AppDetail> {

    override fun compare(left: AppDetail?, right: AppDetail?): Int {
        return left?.name?.lowercase()?.compareTo(right?.name?.lowercase()!!)!!
    }
}